platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/validator/LogMarkerFilter.java

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.otter.fixtures.validator;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Set;
+import org.apache.logging.log4j.Marker;
+import org.hiero.otter.fixtures.Validator.LogFilter;
+import org.hiero.otter.fixtures.logging.StructuredLog;
+
+/**
+ * A {@link LogFilter} that filters log messages based on their {@link Marker}.
+ *
+ * <p>This filter allows inclusion or exclusion of log messages depending on whether
+ * their marker is present in a given set of markers.
+ *
+ * @param include if {@code true}, messages whose marker is <b>not</b> in the set are filtered out;
+ *                if {@code false}, messages whose marker <b>is</b> in the set are filtered out
+ * @param markers the set of markers used for filtering
+ */
+public record LogMarkerFilter(boolean include, @NonNull Set<Marker> markers) implements LogFilter {
```

Comment:
Nit: I prefer enums with two values and specific names instead of `boolean` in such cases.